testhelpers/steps.go

```diff
@@ -314,6 +314,16 @@ func (ctx *TestContext) TheResponseBodyShouldBeJSON(body *gherkin.DocString) (er
 		return
 	}
 
+	for i, b := range actual {
+		if b != expected[i] {
```

Comment:
Why moving it here before checking their length? If their lengths are not matchings, they may overflow in this loop.

Reply:
I needed to do this to see how the strings differ. The message about lengths is not very informative

Reply:
The point is that may do an overflow if `expected` is shorter than `actual` 🙂